### ISSUE TITLE
403 Forbidden on MOVE inside a shared folder

### ISSUE THREAD

**Issue by elettrico** [nextcloud/server#5418]
Hi, I have a working installation of Nextcloud 12. I have this configuration:
- user A: linux with nextcloud client
- user B: linux with nextcloud client
- admin
with admin I've created user A and user B, put them in group "sistema" and then created a folder shared with the same group. Sharing files and syncronizing works fine, but not moving files inside the shared folder. 

### Steps to reproduce
1. user A create folder "sistema/F1"
2. user A create file "sistema/text.txt"
3. user A move "sistema/text.txt" to ""sistema/F1/text.txt"

### Expected behaviour
Nextcloud client synchronize the move to the server

### Actual behaviour
Nextcloud client tell me he cannot synchronize "text.txt" because of an "access forbidden" error.
The same procedure outside the shared folder (ie: personal user folder) works fine.

### Server configuration

**Operating system**: Centos 6.8, dedicated server with WHM/CPanel
**Web server:** Apache 2.4.25
**Database:** Mysql
**PHP version:** PHP 5.6.30 (cgi-fcgi)
**Nextcloud version:** 12
**Updated from an older Nextcloud/ownCloud or fresh install:** updated from a fresh nextcloud 11 install
**Where did you install Nextcloud from:** sources
**Signing status:**
<details>
<summary>Signing status</summary>
```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
```
Enabled:
  - activity: 2.5.2
  - bruteforcesettings: 1.0.2
  - comments: 1.2.0
  - dav: 1.3.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>
```
{
    "system": {
        "instanceid": "oce2paasocgs",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.ttt.com"
        ],
        "datadirectory": "\/home\/ttt\/nextcloud.data",
        "overwrite.cli.url": "http:\/\/cloud.ttt.com",
        "dbtype": "mysql",
        "version": "12.0.0.29",
        "dbname": "ttt_nextcloud",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "asset-pipeline.enabled": true,
        "maintenance": false,
        "theme": "",
        "loglevel": 2
    }
}
```
</details>

**Are you using external storage, if yes which one:** no
**Are you using encryption:** no
**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Nextcloud linux client
**Operating system:** linux

#### Web server error log
<details>
<summary>Web server error log</summary>
```
"MOVE /remote.php/dav/files/usera/sistemas/text.txt HTTP/1.1" 403 192 "-" "Mozilla/5.0 (Linux) mirall/2.3.1 (Nextcloud)"
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
no log is generated about the move operation
```
</details>

**Comment by elettrico** [comment]
(I've changed issue title because I've tried without forcing https and the issue is still here, so it's not the problem)

**Comment by MorrisJobke** [comment]
Duplicate of #3502

Fixed in the upcoming 12.0.1
